src/compiler/utilities.ts

```diff
@@ -4129,13 +4154,16 @@ namespace ts {
         return stringContains(version, "-dev") || stringContains(version, "-insiders");
     }
 
-    export function createTextWriter(newLine: string): EmitTextWriter {
+    export function createTextWriter(newLine: string, indentation?: string | number | boolean): EmitTextWriter {
         let output: string;
         let indent: number;
         let lineStart: boolean;
         let lineCount: number;
         let linePos: number;
         let hasTrailingComment = false;
+        if (indentation !== undefined) {
+            setIndentString(indentation);

```

Comment:
The implication of this is that if one writer is created with a specific indent, all following writers will have that indent if they don't explicitly set that, and also, creating a new writer with a different indent will change the indent for all writers that are still living.

Reply:
yes... 100% correct... as the description in the PR attempted to say.  

Consideration:  Depreciating the existing ts.getIndentString and ts.getIndentSize methods may be problematic as although they will be depreciated it does not stop someone from using them... even if they use the new indentation property via ts.createTextWriter.  If they do use the new indentation implementation then it is possible the results from those old methods may not match those setup at instantiation time.

I'll take a stab at making the indentation part of the ts.createTextWriter instance.

Reply:
If they are already using a new API to set the indent, then I doubt they're going to be using the old API as well.